src/gui/qgsfilewidget.cpp

```diff
@@ -231,25 +231,40 @@ void QgsFileWidget::openFileDialog()
 
   // Handle Storage
   QString fileName;
+  QStringList fileNames;
   QString title;
   if ( mStorageMode == GetFile )
   {
     title = !mDialogTitle.isEmpty() ? mDialogTitle : tr( "Select a file" );
     fileName = QFileDialog::getOpenFileName( this, title, QFileInfo( oldPath ).absoluteFilePath(), mFilter );
   }
+  else if ( mStorageMode == GetMultipleFiles )
```

Comment:
Could you swap all these ifs to switches? (With no default cases). It makes maintenance easier.